### PR TITLE
Update to Python 3.12

### LINKED ISSUE #32516: [$250 bounty] Update to Python 3.12
This is a pre-requisite to updating to Ubuntu 24.04.

Here's a starting point: https://github.com/commaai/openpilot/pull/32296.

Main blocker is upstreaming 3.12 support to MetaDrive: https://github.com/metadriverse/metadrive.

### PR DESCRIPTION
**Pull request by BBBmau**
Fixes #32516

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [project]
 name = "openpilot"
-requires-python = ">= 3.11"
+requires-python = ">= 3.12"
```
Comment (adeebshihadeh) [inline pyproject.toml:3]:
```suggestion
requires-python = ">= 3.11"
```

**Comment by BBBmau** [comment]
For context: I went ahead and created a python package from my metadrive PR just so we don't have to wait. We can change it to be the original metadrive package once a new release is made with it.

https://pypi.org/project/metadrive-simulator-py3-12/0.2.0/

**Comment by BBBmau** [comment]
This seems like a common issue with moving to 3.12: `ModuleNotFound Error: No module named `six.moves``

I see a lot of mentions and how using 3.11 fixed it 😕 

https://github.com/Datalux/Osintgram/issues/947#issuecomment-1872485426

**Comment by BBBmau** [comment]
removing `--system-site-packages` solved the previous issue that I was having where `six` modules were not being found.

**Comment by BBBmau** [comment]
closing since this relies on pyenv being removed. Making a new PR that adds support with pyenv present.

**Pull request closed without merging**